### ISSUE TITLE
transient-define-prefix autoload is broken with current transient version

### ISSUE THREAD

**Issue by TamsynUlthara** [rjekker/password-store-menu#2]
The autoload for `(transient-define-prefix password-store-menu)` is broken in the manner described in https://github.com/magit/transient/issues/280 when using the most recent transient from MELPA.

Specifically, `(package-initialize)` results in: `Error loading autoloads: (void-function transient-prefix)`

The solution looks like it's documented in this commit: https://github.com/magit/transient/commit/0dd64387d3847d63298fbe10bf73f20aeef0cf2a

> **How can I autoload prefix and suffix commands?**
>
> If your package only supports Emacs 30, just prefix the definition
> with `;;;###autoload`.  If your package supports released versions of
> Emacs, you unfortunately have to use a long form autoload comment
> as described in [Autoload](https://www.gnu.org/software/emacs/manual/html_node/elisp/Autoload.html).
>
> ```emacs-lisp
>   ;;;###autoload (autoload 'magit-dispatch "magit" nil t)
>   (transient-define-prefix magit-dispatch ()
>     ...)
> ```

That didn't seem to actually solve the problem here, but commenting out this line in `password-store-menu-autoloads.el` seems to prevent the error:

```emacs-lisp
(put 'password-store-menu 'transient--prefix (transient-prefix :command 'password-store-menu))
```

**Comment by rjekker** [comment]
I cannot reproduce this problem. Which version of emacs are you using?

**Comment by danielsz** [comment]
I can confirm the problem. I am on Emacs 29.4.
It's an autoload problem. After upgrading packages with the built-in `package.el`, `package-quickstart.el` and `package-quickstart.elc` are being generated and henceforth emacs cannot start.

`Error loading autoloads: (void-function transient-prefix)`

The workaround is to delete the generated files (which concatenates all the autoload directives in one file), but this really should be fixed here.

**Comment by rjekker** [comment]
@danielsz please let me know if my fix for this issue actually solves the problem.

**Comment by danielsz** [comment]
Yes, this certainly seems to fix the problem. Thank you so much!

And thank you for this package! I appreciate it.
